app/controllers/subject_rules_controller.rb

```diff
@@ -0,0 +1,72 @@
+class SubjectRulesController < ApplicationController
+  def index
+    authorize workflow
+
+    rules = SubjectRule.where(workflow_id: params[:workflow_id])
+    respond_to do |format|
+      format.html { redirect_to workflow }
+      format.json { render json: rules }
+    end
+  end
+
+  def show
+    authorize workflow
+    @rule = workflow.subject_rules.find(params[:id]) or not_found
+    respond_with @rule
+  end
+
+  def new
+    authorize workflow
+    @rule = SubjectRule.new(workflow: workflow)
+  end
+
+  def edit
+    authorize workflow
+    @rule = SubjectRule.find(id: params[:id]) or not_found
+  end
+
+  def create
+    authorize workflow
+
+    @rule = SubjectRule.new(rule_params)
+    @rule.save
+
+    respond_to do |format|
+      format.html{ redirect_to workflow }
+      format.json{ render json: @rule}
+    end
+  end
+
+  def update
+    authorize workflow
+    @rule = workflow.subject_rules.find(params[:id]) or not_found
+
+    if @rule.update(rule_params)
+      respond_to do |format|
+        format.html { redirect_to workflow, success: 'Rule updated' }
+        format.json { render json: @rule }
+      end
+    else
+      respond_with @rule
+    end
+  end
+
+  def destroy
+    authorize workflow
+    rule = workflow.subject_rules.find(params[:id])
+
+    rule.destroy
+    respond_with rule, location: [workflow]
+  end
+
+  private
+
+  def workflow
+    @workflow ||= policy_scope(Workflow).find(params[:workflow_id])
+  end
+
+  def rule_params
+    p = params.require(:subject_rule).permit(:id)
+    p.merge(condition: params[:subject_rule][:condition], workflow_id: workflow.id)
```

Comment:
I know you struggled to get this working. I've done some debugging and figured it out, and figured I might as well commit those directly. The core fix was basically that `format: :json` in the tests will set the `Accept: application/json` header, but if used on a `post` or `put`, it will annoyingly not *send* the body as JSON. Figured this out by prying into this line and looking at `request.body.read` and seeing the stuff as CGI escaped querystring. 

RSpec has a separate `as: :json` which makes it convert params to JSON and then send it as `Content-Type: application/json` which was the fix. 

This has bitten me more times than it should have, and still I forget the difference between the two every time. :sadpanda:

Reply:
oh wow, i never knew that--thank you for the explanation!